### PR TITLE
Int 553 validate names for collateral

### PR DESCRIPTION
**Pull request by LotharII**
### Jira Story

- [Validate names for collateral INT-553](https://osi-cwds.atlassian.net/browse/INT-553)

### Purpose
First Name and Last Name should be required while roles includes 'Collateral' role.

### CODE REVIEW

File: spec/features/screening/validations/person_validation_spec.rb [hunk 1]
```diff
@@ -58,8 +58,44 @@
       end
     end
 
-    context 'first name or last name not entered while roles does not include Victim role' do
-      scenario 'error not displayed' do
+    context 'first name or last name not entered while roles includes Collateral role' do
```
Comment (NagaPradeepGajula-osi) [inline spec/features/screening/validations/person_validation_spec.rb]: I am little confused of having the same context and scenario with different tests at 61 and 184. Can you rewrite these clearly.

File: spec/features/screening/validations/person_validation_spec.rb [hunk 2]
```diff
@@ -144,5 +180,69 @@
         end
       end
     end
+
+    context 'first name or last name not entered while roles includes Collateral role' do
```
Comment (NagaPradeepGajula-osi) [inline spec/features/screening/validations/person_validation_spec.rb]: same as 61 context.

File: spec/features/screening/validations/person_validation_spec.rb (hunk 1 above)
Comment (LotharII) [reply spec/features/screening/validations/person_validation_spec.rb]: There are different roles. Victim and Collateral.

File: app/javascript/selectors/screening/personShowSelectors.js [hunk 3]
```diff
@@ -15,28 +15,32 @@ export const getErrorsSelector = (state, personId) => {
...
-  return roles.includes('Victim')
+  return (roles.includes('Victim') || roles.includes('Collateral'))
 }
 
 export const getPersonAlertErrorMessageSelector = (state, personId) => {
   const required = getNamesRequiredSelector(state, personId)
   const person = state.get('participants').find((person) => person.get('id') === personId) || Map()
   const lastName = person.get('last_name')
   const firstName = person.get('first_name')
+  const roles = person.get('roles', List())
```
Comment (NagaPradeepGajula-osi) [inline app/javascript/selectors/screening/personShowSelectors.js]: code climate is complaining about Cognitive Complexity of 6 (exceeds 5 allowed). Look for refactoring this.

File: spec/features/screening/validations/person_validation_spec.rb (hunk 1 above)
Comment (LotharII) [reply spec/features/screening/validations/person_validation_spec.rb]: There are different root contexts: ' When page is opened in edit mode' at 61 and 'When page is opened in show mode' at 184.

File: app/javascript/selectors/screening/peopleFormSelectors.js [hunk 4]
```diff
@@ -133,11 +133,15 @@ export const getPersonAlertErrorMessageSelector = (state, personId) => {
   const required = getNamesRequiredSelector(state, personId)
   const lastName = state.getIn(['peopleForm', personId, 'last_name', 'value'])
   const firstName = state.getIn(['peopleForm', personId, 'first_name', 'value'])
+  const roles = state.getIn(['peopleForm', personId, 'roles', 'value'], List())
```
Comment (NagaPradeepGajula-osi) [inline app/javascript/selectors/screening/peopleFormSelectors.js]: code climate is complaining about Cognitive Complexity of 6 (exceeds 5 allowed). Look for refactoring this.

File: spec/features/screening/validations/person_validation_spec.rb (hunk 1 above)
Comment (NagaPradeepGajula-osi) [reply spec/features/screening/validations/person_validation_spec.rb]: 👍

File: spec/features/screening/validations/person_validation_spec.rb (hunk 2 above)
Comment (NagaPradeepGajula-osi) [reply spec/features/screening/validations/person_validation_spec.rb]: 👍

**Pull request merged**